### PR TITLE
Migrate CircleCI to Github actions

### PR DESCRIPTION
**Pull request by leplatrem**
- [x] Port config files
- [ ] Verify actions behaviour
- [x] Configure Dockerhub secrets
- [ ] Delete project from CircleCI
- [ ] Test container publication

### CODE REVIEW

File: .github/workflows/build-publish.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+jobs:
+  build-and-publish:
+    runs-on: ubuntu-latest
+    env:
+      TEST_TAG: ${{ github.repository }}:test
+      TEST_CONTAINER_NAME: container-healthcheck
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
```
Comment (grahamalama) [inline .github/workflows/build-publish.yaml:21]: I noticed checkout@v4 recently came out. Might as well upgrade here?

File: .github/workflows/build-publish.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      - name: Set tag version
+        run: echo "VERSION_TAG=$(git describe --tags --abbrev=4)" >> $GITHUB_ENV
+      - name: Build `version.json` file
+        run: |
+          printf '{\n    "commit": "%s",\n    "version": "%s",\n    "source": "%s",\n    "build": "%s"\n}\n' \
+            "$GITHUB_SHA" \
+            "$VERSION_TAG" \
+            "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY" \
+            "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" > ./version.json
+
```
Comment (grahamalama) [inline .github/workflows/build-publish.yaml:33]: Should we cat the version file here for debugging?

File: .github/workflows/build-publish.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      - name: Check that container is running
+        run: |
+          docker exec ${{ env.TEST_CONTAINER_NAME }} curl --retry 10 --retry-delay 1 --retry-connrefused http://0.0.0.0:8000/checks | grep remotesettings
+
+      - name: Spin down container
+        run: |
+          docker rm -f ${{ env.TEST_CONTAINER_NAME }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v3
```
Comment (grahamalama) [inline .github/workflows/build-publish.yaml:87]: Does this rebuild the container from scratch? Or somehow use the container built in the previous `build-push-action`?

Not something to handle in this PR, but if it's the former, maybe we can add an issue to cache the container so we don't have to rebuild it.

File: .github/workflows/build-publish.yaml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        run: |
+          printf '{\n    "commit": "%s",\n    "version": "%s",\n    "source": "%s",\n    "build": "%s"\n}\n' \
+            "$GITHUB_SHA" \
+            "$VERSION_TAG" \
+            "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY" \
+            "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" > ./version.json
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
```
Comment (grahamalama) [inline .github/workflows/build-publish.yaml:36]: Could use `docker/metadata-action@v5`

File: .github/workflows/build-publish.yaml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+      - name: Login to Docker Hub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and export to Docker
+        uses: docker/build-push-action@v3
```
Comment (grahamalama) [inline .github/workflows/build-publish.yaml:53]: Could use `docker/build-push-action@v5`

**Review by grahamalama — APPROVED** [review]
Nice!

File: .github/workflows/build-publish.yaml (hunk 3 above)
Comment (leplatrem) [reply .github/workflows/build-publish.yaml]: https://github.com/mozilla-services/telescope/issues/1332

**Pull request merged**